src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -754,3 +763,76 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal, II
 		this.raw.write(data);
 	}
 }
+
+class AccessibleBuffer extends DisposableStore {
+
+	private _accessibilityBuffer: HTMLElement | undefined;
+	private _bufferElementFragment: DocumentFragment | undefined;
+
+	constructor(
+		private readonly _terminal: RawXtermTerminal,
+		private readonly _font: ITerminalFont,
+		private readonly _capabilities: ITerminalCapabilityStore,
+		@IAccessibilityService private readonly _accessibilityService: IAccessibilityService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService
+	) {
+		super();
+		this.add(this._terminal.registerBufferElementProvider({ provideBufferElements: () => this.focus() }));
+	}
+
+	focus(): DocumentFragment {
+		if (!this._bufferElementFragment) {
+			this._bufferElementFragment = document.createDocumentFragment();
+		}
```

Comment:
This doesn't technically solve https://github.com/microsoft/vscode/issues/173734 but my main goal was to encapsulate things a little more which is accomplished via `AccessibleBuffer`

Reply:
🤔 but we're updating the elements in the callback https://github.com/microsoft/vscode/blob/062ecb0e9657cc5d441135ea5701f9059f7099f2/src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts#L780-L785

Reply:
You might need to do this to get the screen reader to read it correctly, but this is still a little weird as this means the command triggers `focusAccessibilityBuffer`, which calls into `AccessibleBuffer.focus` and generates the fragment, then xterm.js requests buffer elements and then calls `AccessibleBuffer.focus` again. So this is a little dangerous as it's recursive, I think the only reason it's working fine is calling focus on a focused element won't trigger the `focus` event listener.

I was imagining:

- User focuses the element (either via our command or manually somehow)
- xterm.js listens to focus on the element which calls `provideBufferElements`, this does not focus, it only provides the elements

This is simpler with no recursion.